lib/repository.py

```diff
@@ -134,17 +134,10 @@ def checkout(self, ref_name):
         Check out the reference name, resetting the index state.
         The reference may be a branch, tag or commit.
         """
-        LOG.info("%(name)s: Fetching repository remotes"
-                 % dict(name=self.name))
-        for remote in self.remotes:
-            try:
-                remote.fetch()
-            except git.exc.GitCommandError:
-                LOG.debug("Failed to fetch %s remote for %s"
-                          % (remote.name, self.name))
-                pass
-            else:
-                LOG.info("Fetched changes for %s" % remote.name)
+        LOG.info("%(name)s: Fetching repository remote %(remote)s"
+                 % dict(name=self.name, remote=MAIN_REMOTE_NAME))
+        main_remote = self.remote(MAIN_REMOTE_NAME)
+        main_remote.fetch()
```

Comment:
Do you mean leaving the debug message and re-raising the exception or allowing the command to fail?
We allowed the command to fail previously because we didn't know which was the remote that should be fetched and some of them might be unaccessible or invalid. Now that we only the "origin" remote, whose URL and refspecs are configurable by the user, I don't see a reason to let it fail. The default options should *never* fail, even if the URL is changed and PR refspecs are not found.